cincinnati/src/lib.rs

```diff
@@ -237,14 +238,92 @@ impl Serialize for Graph {
     }
 }
 
+impl PartialEq for Graph {
+    fn eq(&self, other: &Graph) -> bool {
+        use daggy::petgraph::visit::IntoNeighbors;
+
+        let asc_order_release_by_version = {
+            use std::cmp::Ordering::{self, *};
+
+            |a: &&Release, b: &&Release| -> Ordering {
+                if a.version() < b.version() {
+                    Less
+                } else if a.version() == b.version() {
+                    Equal
+                } else {
+                    Greater
+                }
+            }
+        };
+
+        // Look through all nodes in self
+        self.dag.node_references().all(|node_ref| {
+            let dag_other = &other.dag;
+            let node_index = node_ref.0;
+            let release = node_ref.1;
+
+            // For each node in self, look through all nodes in other and find a match
+            dag_other
+                .node_references()
+                .filter(|node_ref_other| {
+                    let node_index_other = node_ref_other.0;
+                    let release_other = node_ref_other.1;
+
+                    // Ensure the set of neighbors of release and release_other are identical
+                    let compare_neighbors = || {
+                        let (neighbors_count, neighbors_other_count) = (
+                            self.dag.neighbors(node_index).count(),
+                            dag_other.neighbors(node_index_other).count(),
+                        );
+
+                        if neighbors_count != neighbors_other_count {
+                            return false;
+                        }
+
+                        let mut neighbors = self
+                            .dag
+                            .neighbors(node_index)
+                            .zip(dag_other.neighbors(node_index_other))
+                            .fold(
```

Comment:
Side note: we are three level of loop-nesting at this point, all in serial logic. I think it is fine for now in order to start landing an `Eq` implementation. Longer term we may (or may not) have a need to investigate a more efficient DAG comparison algorithm.